SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -500,6 +489,53 @@ void FLocalDeploymentManager::FinishLocalDeploymentShutDown()
 	bStoppingDeployment = false;
 }
 
+void FLocalDeploymentManager::WindowsRequestGracefulRuntimeShutdown()
+{
+	// Sends a process signal requesting the runtime shutdown
+
+	// Prepare runtime process name
+	FString RuntimeProcessName = RuntimePath;
+	RuntimeProcessName = RuntimeProcessName.Replace(TEXT("/"), TEXT("\\"));
+	// Find runtime window
+	const HWND RuntimeWindowHandle = FindWindowA(nullptr, TCHAR_TO_ANSI(*RuntimeProcessName));
+	if (RuntimeWindowHandle != nullptr)
+	{
+		// Send message to runtime window to close gracefully.
+		SendMessage(RuntimeWindowHandle, WM_CLOSE, NULL, NULL);
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error,
+			   TEXT("Tried to gracefully stop local deployment but could not find runtime window. Runtime will eventually stop "
+					"non-gracefully."));
+	}
+}
+
+void FLocalDeploymentManager::NonWindowsRequestGracefulRuntimeShutdown()
+{
+	// Sends a http message requesting graceful shutdown
```

Comment:
Why don't we do any of this stuff on windows anymore?

Reply:
Also we don't try close any windows on non windows platforms? Do we just not care?

Reply:
process shutdown is slightly faster, but implementation is different on mac (i.e. nonwindows). Will leave a comment in the code on this

Reply:
r.e. second question, the essence of what we're doing here is gracefully shutting down the runtime. not actually sure why we're doing that via a 'window' I got the code from @MatthewSandfordImprobable (he previously had a look at doing this but other things then meant it didn't work). Would be interested why we're actually doing this matt

Reply:
oh, having read a bit more about windows signals it seems like you need a window handle to call WM_CLOSE (which is nicer than terminateprocess)

Reply:
If we have a way to close the process on mac super! I had not luck when I was searching. I tried some non platform specific methods to close the process but they all seemed to terminate the process rather than send a close signal.

Reply:
@MatthewSandfordImprobable I should've asked actually, is the runtime listening for a SIGTERM signal or something on unix? I don't really know how WM_CLOSE maps to unix